oerp-pricelist.py

```diff
@@ -263,28 +274,25 @@ def import_products_oerp(data, extra_filters=None, columns=None):
                     "manufacturer_pref",
                     "seller_ids",
                     "property_stock_location"]
-    print "OERP Import"
     prod_ids = oerp.search('product.product', [('default_code', '!=', False)] + extra_filters)
-    print "reading {} products from OERP, this may take some minutes...".format(len(prod_ids))
     prods = []
 
     def split_list(prod_list, chunk_size):
         return [prod_list[offs:offs + chunk_size] for offs in range(0, len(prod_list), chunk_size)]
 
     # columns starting with _ are generated in this script and not from the DB
     query_columns = [col for col in columns if not col.startswith("_")]
-    # read max. n products at once
-    n = 100
-    for prod_ids_slice in split_list(prod_ids, n):
-        print "."
-        prods += oerp.read('product.product', prod_ids_slice, query_columns,
-                           context=oerp.context)
-        time.sleep(2)
```

Comment:
The sleep call must remain to avoid flooding the ERP server. Without this fix, we had bad experience with the usability of the ERP website while the script is running.

Reply:
I'd hazard the guess that with the much improved caching this won't hit nearly as hard as before. Runtime feels an order of magnitude faster now than before. A sleep of 2 seconds will _significantly_ reduce performance which is annoying during development.
The proper solution would be, e.g., a leaky bucket through which all API calls go first. This would eliminate all bursts. and thus prevent overloading the server and improve usability of the website.
The hacky and much easier solution would be to have that two second sleep but only if it runs on the server (as part of a cron job, systemd timer, you name it). An environment variable comes to mind for setting the sleep duration, defaulting to the two seconds.
Thoughts?

Reply:
I went ahead and implemented rate limting defaulting to one request every two seconds. It can easily be changed via two environment variables (see README.MD)

Reply:
Note that the requests don't all have the same "cost", it depends on how much (e.g., how many products) you request at once. The new ratelimiting may be stricter than the old sleep, because it also considers other requests than just the single problematic one which reads 100 products. Why don't we just keep the old way for the cronjob, with an option to disable it during development?

I don't see the point of all the added complexity for a script that will be probably dropped in a few years.

Reply:
I agree! I went for rate limiting because it more accurately models the problem and it was, in fact, a very simple change. The hardest part was finding a library that worked with the old Python version.

As you probably surmised, I assumed the costliest part to be the network I/O overhead, not the actual domain logic behind it. In my experience the former typically contributes a lot more to overall load and runtime, hence the universal rate limit. With this not applying in this instance, a change to the PR is, of course, in order!